Choropleth Docs

Kully opened the pull request:
todo:
- [x] I think https://plot.ly/python/choropleth-maps/ should have the USA county example at the bottom with a link to the full county choropleth docs, just in case people forget to search “county” when they search for choropleth maps (Jack's comment)
- [x] https://plot.ly/python/county-level-choropleth/ move to the URL python/mapbox-county-choropleths

Kully commented:
@cldougl whenever you want to have a look. Still some todos in this PR but the doc is written.

cldougl commented:
@Kully if you move https://plot.ly/python/county-level-choropleth/ to `python/mapbox-county-choropleths` can you redirect `python/county-level-choropleth/` to `python/county-choropleth/` (ie the new doc)

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+</div>
+<div class="inner_cell">
+<div class="text_cell_render border-box-sizing rendered_html">
+<h4 id="Change-your-Scope">Change your Scope<a class="anchor-link" href="#Change-your-Scope">&#194;&#182;</a></h4><p>Even if your FIPS values belong to a single state, changing the <code>scope</code> of the choropleth shifts the zoom and position of the USA map to include any states. Specify a list of state names and the camera automatically adjusts to include the state outlines of the selected s ...
```
>Even if your FIPS values belong to a single state, changing the scope of the choropleth shifts the zoom and position of the USA map to include any states. Specify a list of state names and the camera automatically adjusts to include the state outlines of the selected states.

Even if your FIPS values belong to a single state, the `scope` defaults to the entire United States as displayed in the example above. Changing the scope of the choropleth shifts the zoom and position of the USA map. You can define the `scope` with a list of state names and the zoom will automatically adjust to include the state outlines of the selected states.

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+</div>
+<div class="inner_cell">
+<div class="text_cell_render border-box-sizing rendered_html">
+<h4 id="Florida">Florida<a class="anchor-link" href="#Florida">&#194;&#182;</a></h4>
```
instead of the name of the state- let's make this title more informative:
### Single State

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+</div>
+<div class="inner_cell">
+<div class="text_cell_render border-box-sizing rendered_html">
+<h4 id="Offline-Choropleth">Offline Choropleth<a class="anchor-link" href="#Offline-Choropleth">&#194;&#182;</a></h4><p>Setting <code>offline_mode=True</code> modifies the centroid markers so that they are only visible if you hoverdrag a box over them.</p>
```
how does this `offline_mode` relate to `plotly.offline`? That should be clarified in this section. As well as linking to the getting started offline doc section: https://plot.ly/python/getting-started/#initialization-for-offline-plotting

Kully replied:
See https://github.com/plotly/plotly.py/issues/944

cldougl replied:
cool! I'd just remove this from the doc or change it to an online example if you still want to include this example.

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+
+
+<div class="output_html rendered_html output_subarea ">
+<div id="b3400adb-0882-4efa-9ade-a51aa48e6c32" style="height: 450px; width: 900px;" class="plotly-graph-div"></div><script type="text/javascript">require(["plotly"], function(Plotly) { window.PLOTLYENV=window.PLOTLYENV || {};window.PLOTLYENV.BASE_URL="https://plot.ly";Plotly.newPlot("b3400adb-0882-4efa-9ade-a51aa48e6c32", [{"name": 1424, "fillcolor": "#8dd3c7", "mode": "lines", "hoverinfo": "text ...
```
it seems like a lot of the options here are set to their default?
I would double check these and remove/simplify any attributes that are set to their default, for example:
isn't the scope default 'usa'?

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+
+<span class="n">fig</span> <span class="o">=</span> <span class="n">ff</span><span class="o">.</span><span class="n">create_choropleth</span><span class="p">(</span>
+    <span class="n">fips</span><span class="o">=</span><span class="n">fips</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="n">values</span><span class="p">,</span> <span class="n">scope</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;CA&#39;</span><span class="p">,</span> <span class="s1">&#39;AZ&#39;</span><span class= ...
+    <span class="n">binning_endpoints</span><span class="o">=</span><span class="p">[</span><span class="mi">14348</span><span class="p">,</span> <span class="mi">63983</span><span class="p">,</span> <span class="mi">134827</span><span class="p">,</span> <span class="mi">426762</span><span class="p">,</span> <span class="mi">2081313</span><span class="p">],</span> <span class="n">colorscale</span ...
```
would be good to explain what binning endpoints is

Kully commented:
> @Kully if you move https://plot.ly/python/county-level-choropleth/ to python/mapbox-county-choropleths can you redirect python/county-level-choropleth/ to python/county-choropleth/ (ie the new doc)

I think `python/mapbox-county-choropleth` is better than `python/mapbox-county-choropleths`. Thoughts?

Kully commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+</div>
+<div class="inner_cell">
+<div class="text_cell_render border-box-sizing rendered_html">
+<h4 id="Version-Check">Version Check<a class="anchor-link" href="#Version-Check">&#194;&#182;</a></h4><p>Note: <code>County Choropleths</code> are available in version <b>2.4.1+</b><br>
```
<3

Kully commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+</div>
+<div class="inner_cell">
+<div class="text_cell_render border-box-sizing rendered_html">
+<h4 id="Version-Check">Version Check<a class="anchor-link" href="#Version-Check">&#194;&#182;</a></h4><p>Note: <code>County Choropleths</code> are available in version <b>2.4.1+</b><br>
```
wdyt about changing the doc name to `USA County Choropleths` then? Just the title, not the url

cldougl replied:
good idea!

cldougl replied:
I would change both - title and Note: US County Choropleths are available in version 2.4.1+

(just want to make this clear as we get asked pretty often about choropleths for other regions outside of the US)

Kully commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,796 @@
+
+<span class="n">fig</span> <span class="o">=</span> <span class="n">ff</span><span class="o">.</span><span class="n">create_choropleth</span><span class="p">(</span>
+    <span class="n">fips</span><span class="o">=</span><span class="n">fips</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="n">values</span><span class="p">,</span>
+    <span class="n">county_outline</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;color&#39;</span><span class="p">:</span> <span class="s1">&#39;rgb(255,255,255)&#39;</span><span class="p">,</span> <span class="s1">&#39;width&#39;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">},</span>
```
good idea

cldougl commented:
@Kully sure `python/mapbox-county-choropleth` is fine 👍  -- the main point is to redirect the old url: https://plot.ly/python/county-level-choropleth/ to the new url: https://plot.ly/python/county-choropleth/

Kully commented:
@cldougl whenever you have time for a 2nd 👀 

NB I removed `example_index` from the mapbox (just mapbox) doc to ensure 5 thumbnails in the maps section and also tried to correct the redirect url in the html file I added.

Can you check if my syntax is correct?

cldougl commented:
there should be only 4 thumbnails on this page. I think we only need 1 choropleth on the index page. I would probably just show county on the maps page (ie more maps) but if you want to leave it there to showcase a new feature then remove the choropleth map option (lines on maps and scatter on mapbox should remain on this index page)
![screen shot 2018-02-21 at 5 08 04 pm](https://user-images.githubusercontent.com/2789078/36508146-dc2b7670-1729-11e8-97c0-6b8cce6ff03b.png)

cldougl commented:
I think the titles should be:
USA County Choropleth Maps // Choropleth Maps // Mapbox Choropleth Maps
![screen shot 2018-02-21 at 5 15 53 pm](https://user-images.githubusercontent.com/2789078/36508514-ec044a76-172a-11e8-8649-ff0f0a220cfa.png)

cldougl commented:
looks like the `county-level-choropleth` folder is now empty- if that's the case can you remove it all-together (it will make this map folder cleaner :) )?

cldougl commented on _posts/python/maps/choropleth-maps/Choropleth_maps.ipynb:
```diff
@@ -276,136 +270,81 @@
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The draw time for this plot will be slow for clients without much RAM.\n"
```
this warning is ok

cldougl commented on _posts/python/maps/choropleth-maps/Choropleth_maps.ipynb:
```diff
@@ -276,136 +270,81 @@
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/plotly/api/v1/clientresp.py:40: UserWarning:\n",
```
can you remove this warning from the docs?
(you can run the cell twice or run: 
```
import warnings
warnings.filterwarnings('ignore')
``` 
in the beginning of your notebook )

cldougl commented on _posts/python/maps/choropleth-maps/Choropleth_maps.ipynb:
```diff
@@ -418,14 +357,8 @@
-    "collapsed": false
-   },
+   "execution_count": 5,
+   "metadata": {},
```
hmm looks like there's no reference section in this doc- can you add a reference section pointing to: https://plot.ly/python/reference/#choropleth

Kully commented:
> looks like the county-level-choropleth folder is now empty- if that's the case can you remove it all-together (it will make this map folder cleaner :) )?

I don't see the folder in my local branch. Where do you see the folder?

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,762 @@
+<div class="text_cell_render border-box-sizing rendered_html">
+<h4 id="FIPS-and-Values">FIPS and Values<a class="anchor-link" href="#FIPS-and-Values">&#194;&#182;</a></h4><p>Every US state and county has an assined ID regulated by the US Federal Government under the term FIPS (Federal Information Processing Standards) codes. There are state codes and county codes: the 2016 state and county FIPS codes can be found at the <a href="https://www.census.gov/geogra ...
+<p>Combine a state FIPS code (eg. <code>06</code> for California) with a county FIPS code of the state (eg. <code>059</code> for Orange county) and this new state-county FIPS code (<code>06059</code>) uniquely refers to the specified state and county.</p>
+<p><code>ff.create_choropleth</code> only needs a list of FIPS codes and a list of values.</p>
```
can you combine this sentence with the one below- there's a lot of space when the doc is rendered:


![screen shot 2018-02-21 at 5 29 55 pm](https://user-images.githubusercontent.com/2789078/36509228-ffa4f9a2-172c-11e8-9d69-6f0f836a712c.png)

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,762 @@
+<p>By default <code>scope</code> is set to <code>['USA']</code> which the API treats as identical to passing a list of all 50 state names:<br></p>
+<p><code>['AK', 'AL', 'CA', ...]</code></p>
+<p>State abbreviations (eg. <code>CA</code>) or the proper names (eg. <code>California</code>) as strings are accepted. If the state name is not recognized, the API will throw a Warning and indicate which FIPS values were ignored.</p>
+<p>Another param used in the example below is <code>binning_endpoints</code>. If your <code>values</code> is a list of numbers, you can bin your values into half-open intervals on the real line.</p>
```
ah I see!

cldougl commented on _posts/python/maps/county_choropleth/2015-06-30-county_choropleth.html:
```diff
@@ -0,0 +1,762 @@
+<p><code>['AK', 'AL', 'CA', ...]</code></p>
+<p>State abbreviations (eg. <code>CA</code>) or the proper names (eg. <code>California</code>) as strings are accepted. If the state name is not recognized, the API will throw a Warning and indicate which FIPS values were ignored.</p>
+<p>Another param used in the example below is <code>binning_endpoints</code>. If your <code>values</code> is a list of numbers, you can bin your values into half-open intervals on the real line.</p>
+<p>For example if <code>binning_endpoints = [4, 7]</code> then each <code>x</code> from <code>values</code> will be grouped into one of the following bins:</p>
```
I would remove this example: it renders kind of weird and I think the description above is enough
![screen shot 2018-02-21 at 5 39 16 pm](https://user-images.githubusercontent.com/2789078/36509616-3b342334-172e-11e8-8b14-c9fba2c39be9.png)

Kully commented:
@cldougl all comments addressed

cldougl commented:
💃 💃 💃
